### PR TITLE
T2353

### PR DESCRIPTION
**Pull request by mjschranz**
https://webmademovies.lighthouseapp.com/projects/65733/tickets/2353-add-optional-links-to-popup-and-text-plugins

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @brett-etherworks received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/cbf9eaa13969c21/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/cbf9eaa13969c21/output.txt

Total script time: 0.85 mins

Server is running on http://pr1042.butyr.org/templates/basic/
Run the Test Suite: http://pr1042.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @brett-etherworks received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/09ec0d83e78888f/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/09ec0d83e78888f/output.txt

Total script time: 3.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @brett-etherworks received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/15bc443c2ab8702/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/15bc443c2ab8702/output.txt

Total script time: 2.77 mins

Server is running on http://pr1042.butyr.org/templates/basic/
Run the Test Suite: http://pr1042.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/a345de6161aaee1/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @mjschranz received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/38fe747d2f0a364/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/a345de6161aaee1/output.txt

Total script time: 1.69 mins

Server is running on http://pr1042.butyr.org/templates/basic/
Run the Test Suite: http://pr1042.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/38fe747d2f0a364/output.txt

Total script time: 2.41 mins

File: templates/assets/plugins/popup/popcorn.popup.js [hunk 1]
```diff
@@ -354,22 +360,31 @@
       innerDiv = document.createElement( "div" );
       textContainer = document.createElement( "div" );
 
-      text = text.split( /[\n\r]/ );
```
Comment (jbuck) [inline templates/assets/plugins/popup/popcorn.popup.js]: It looks like you're removing the multi-line text feature here

File: templates/assets/editors/editorhelper.js [hunk 2]
```diff
@@ -123,9 +123,21 @@
...
-      var newText = "";
+      var newText = "",
+          anchorTags,
+          i;
 
       if ( contentContainer ) {
+        anchorTags = contentContainer.querySelectorAll( "a" );
+
+        if ( anchorTags ) {
+          for ( i = 0; i < anchorTags.length; i++ ) {
```
Comment (dseif) [inline templates/assets/editors/editorhelper.js]: `for ( i = 0; i < anchorTags.length; i++ ) {` -> `for ( i = 0, l = anchorTags.length; i < l; i++ ) {`

You will have to create the `l` variable where you create `i` as well.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/24f9f55cd19b64f/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/24f9f55cd19b64f/output.txt

Total script time: 1.07 mins

Server is running on http://pr1042.butyr.org/templates/basic/
Run the Test Suite: http://pr1042.butyr.org/test/

File: templates/assets/plugins/popup/popcorn.popup.js (hunk 1 above)
Comment (mjschranz) [inline templates/assets/plugins/popup/popcorn.popup.js]: Yeah... I noticed that. Don't know why I did.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/2af59d8fb12cba5/output.txt

File: templates/assets/plugins/popup/popcorn.popup.js [hunk 3]
```diff
@@ -359,21 +365,43 @@
...
         }
         node = document.createElement( "span" );
         node.appendChild( document.createTextNode( text[ i ] ) );
-        textContainer.appendChild( node );
+        
+        if ( options.linkUrl ) {
+          link.appendChild( node );
+        } else {
+          textContainer.appendChild( node );
+        }
```
Comment (dseif) [inline templates/assets/plugins/popup/popcorn.popup.js]: these two if blocks can be merged

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/2af59d8fb12cba5/output.txt

Total script time: 3.00 mins

File: templates/assets/plugins/popup/popcorn.popup.js (hunk 3 above)
Comment (dseif) [inline templates/assets/plugins/popup/popcorn.popup.js]: also should store the value of options.linkUrl in a variable since it's accessed so much

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/176d1e63d31f5f7/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/176d1e63d31f5f7/output.txt

Total script time: 3.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/84432f979b4f640/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/49063c04f93ce98/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/49063c04f93ce98/output.txt

Total script time: 3.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/165f10725562d7d/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/165f10725562d7d/output.txt

Total script time: 3.00 mins

File: templates/assets/plugins/popup/popcorn.popup.js (hunk 3 above)
Comment (mjschranz) [inline templates/assets/plugins/popup/popcorn.popup.js]: I need some form of an if check to ensure that I'm not appending a line break the first time through the loop. I can make it a little better but not too much.

File: src/editor/base-editor.js [hunk 4]
```diff
@@ -140,6 +140,38 @@ define( [ "core/eventmanager", "util/scrollbars", "ui/widget/tooltip", "ui/widge
...
+    *                   hover: Triggers if the tooltip displays on hover of element.
+    */
+    extendObject.createTooltip = function( element, options )  {
+      var tooltip;
+
+      options = options || {};
+
+      ToolTip.create( options );
+
+      tooltip = ToolTip.get( options.name );
```
Comment (cadecairos) [inline src/editor/base-editor.js]: if `options.name` is ever undefined we'll have some problems below. Perhaps this fn should not execute at all if there's no options or options.name value.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/a6bce9fb6a5cfe7/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/a6bce9fb6a5cfe7/output.txt

Total script time: 3.00 mins

File: src/editor/base-editor.js [hunk 5]
```diff
@@ -140,6 +140,40 @@ define( [ "core/eventmanager", "util/scrollbars", "ui/widget/tooltip", "ui/widge
...
+    *                   top: The CSS top position of the Tooltip in relation to element.
+    *                   left: The CSS left position of the Tooltip in relation to element.
+    *                   hidden: The Tooltips initial visibility state.
+    *                   hover: Triggers if the tooltip displays on hover of element.
+    */
+    extendObject.createTooltip = function( element, options )  {
+      var tooltip;
+
+      if ( options ) {
+        ToolTip.create( options );
```
Comment (cadecairos) [inline src/editor/base-editor.js]: This probably should only happen if there's a name for the tooltip. ( combine the conditions of the following if statement into the outer if statement. )

**Pull request merged**